verl/experimental/agent_loop/agent_loop.py

```diff
@@ -180,8 +195,26 @@ def __init__(self, config: DictConfig, server_handles: List[ray.actor.ActorHandl
         local_path = copy_to_local(config.actor_rollout_ref.model.path)
         self.tokenizer = hf_tokenizer(local_path, trust_remote_code=True)
 
-        trace_config = config.trainer.get("rollout_trace", {})
+        custom_agent_loop_loader = config.actor_rollout_ref.rollout.agent.custom_agent_loop_loader
+        if custom_agent_loop_loader:
+            module_path, func_name = custom_agent_loop_loader.rsplit(".", 1)
+            module = importlib.import_module(module_path)
+            custom_agent_loop_classes = getattr(module, func_name)()
+            assert isinstance(custom_agent_loop_classes, dict), (
+                "custom_agent_loop_loader should be function: Callable[] -> Dict[str, Type[AgentLoopBase]]"
+            )
```

Comment:
Would you consider using [`hydra.utils.instantiate`](https://hydra.cc/docs/advanced/instantiate_objects/overview/)? It allows passing additional configuration to the AgentLoop, like which graph to use and some config for tools as well ([example here](https://github.com/waleko/verl/blob/7813c0c1798052bedac96cf8cc877ac1eb45647c/examples/grpo_trainer/config/retool_reflection.yaml)). It's really nice.

Reply:
Nice idea, let me try it.

Reply:
@waleko `hydra.utils.instantiate` is supported, please take a look.